KK-DZ/src/app/modules/workers/components/worker-add-edit-dialog/worker-add-edit-dialog.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, ChangeDetectionStrategy, Inject } from "@angular/core";
+import { FormBuilder, FormControl, FormGroup, Validators } from "@angular/forms";
+import { Employee } from "src/app/core/models/employee";
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { EmployeeOrigin } from "src/app/core/models/employee-origin";
+
+@Component({
+    selector:'app-worker-add-edit-dialog',
+    templateUrl: './worker-add-edit-dialog.component.html',
+    styleUrls: ['./worker-add-edit-dialog.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+
+export class WorkerAddEditDialogComponent{
+    public titleText: string;
+    public addForm: FormGroup;
+    public buttonText: string;
+
+    constructor(
+        public dialogRef: MatDialogRef<WorkerAddEditDialogComponent>, 
+        @Inject(MAT_DIALOG_DATA) public data: Employee | null, 
+        private formBuilder: FormBuilder
+    ) {
+        this.titleText = this.data === null ? 'Dodawanie' : `Edytowanie ${this.data.name}`;
+        this.buttonText = this.data === null ? 'Dodaj' : 'Edytuj';
+
+        if(this.data === null) this.data = EmployeeOrigin;
```

Comment:
lepiej nie zmieniać tej zmiennej w konstruktorze, ładniej będzie wyglądać zapis: 
```suggestion
        const worker = this.data || EmployeeOrigin;
```